mars-rover-new/src/mars_rover_new/core.clj

```diff
@@ -1,7 +1,94 @@
+;; Mars Rover Exercise
+
 (ns mars-rover-new.core
+  (:require [clojure.string :as string])
   (:gen-class))
 
+;; todo: input validation, check invalid moves
+
+;; input validation: check for number of args, catch errors during type conversion
+
+(declare add-plateau-size, add-rover-position, add-cmd-sequence)
+
+(def info-map {:plateau-size {:x nil
+                              :y nil}
+               :rover-position {:x nil
+                                :y nil
+                                :direction nil}
+               :cmd-sequence nil})
+
+;; (defn get-input
+;;   "Read one line of input stream. Trim and convert to upper-case."
+;;   []
+;;   (let [input (string/trim (read-line))]
+;;     (string/upper-case input)))
+
+(defn get-input
+  "Prompt (if provided) and read one line of input stream. Trim and convert to upper-case."
+  ([] (get-input ""))
+  ([prompt]
+   (if (not-empty prompt)
+     (println prompt))
+   (let [input (string/trim (read-line))]
+     (string/upper-case input))))
+
+(defn user-input
+  "Reads 3 lines of input from user and calls 3 functions which add this data to info-map"
+  [] 
+  (let [plateau-size (get-input "Enter plateau size:")
+        rover-position (get-input "Enter initial rover position:")
+        cmd-sequence (get-input "Enter command sequence:")]
+    (->> info-map
+         (add-plateau-size plateau-size)
+         (add-rover-position rover-position)
+         (add-cmd-sequence cmd-sequence))))
+
+(defn add-plateau-size
+  "Add plateau-size data to info-map"
+  [plateau-size, info-map]
+  (let [[x y] (map #(Integer/parseInt %) (string/split plateau-size #" "))]
+    (assoc-in info-map [:plateau-size] {:x x
```

Comment:
Consider using `assoc` instead of `assoc-in` since there is no modification of nested keys yet.

Same for `add-rover-position` and `add-cmd-sequence`

Reply:
I think I need to use assoc-in? This is a nested map, not sure how to do this without assoc-in.